### PR TITLE
feat: highest/lowest scoring visitor

### PR DESCRIPTION
**Pull request by austincarrjones**
## Type Of Change
- [X] New Feature
- [ ] Debugging
- [X] Refactor/File Clean-up
- [X] Tests
- [ ] Documentation

## Provide A Summary Of Your Change
closed previous PR because I either accidentally deleted some tests in the game_team_spec (unrelated to this PR anyway) or didn't have main updated properly. 

This one has the 2 methods highest/lowest scoring visitor passing test, but haven't checked out StatTracker yet.

## Group Project Check List
- [X] I made test for my code, and made sure they were passing before submitting PR
- [X] I assigned at least 2 group members to review my PR

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -110,4 +110,53 @@ def self.average_goals_per_season
...
+
+  def self.away_goals_per_team
+    away_goals = Hash.new(0)
+    all_games.each do |game|
+      away_goals[game.away_team_id] += game.away_goals
+    end
+    away_goals
+  end
+
+  def self.away_games_per_team
```
Comment (noahdurbin) [inline lib/game.rb:122]: Nice work on these hashes

**Review by noahdurbin — APPROVED** [review]
Looks Great!

File: spec/game_spec.rb [hunk 2]
```diff
@@ -182,4 +182,55 @@
...
     end
   end
+
+  describe '#away_goals_per_team' do
+    it 'returns hash of teams and total away game goals' do
+      GameFactory.new
+      GameFactory.create_games("./fixtures/games_fixture.csv")
+      
+      expect(Game.away_goals_per_team).to eq({3=>5, 6=>12, 5=>1, 17=>5, 16=>3, 9=>3, 8=>3})
+      GameFactory.reset_games
```
Comment (zach-bergman) [inline spec/game_spec.rb:192]: Tearing down after each it block, nice

File: lib/game.rb [hunk 3]
```diff
@@ -110,4 +110,53 @@ def self.average_goals_per_season
...
+    end
+    team_name_highest
+  end
+
+  def self.lowest_scoring_visitor
+    team_id_lowest = self.away_avg_goals_per_team.min_by { |key, value| value }[0]
+    team_name_lowest = nil
+    teams = TeamFactory.create_teams("./data/teams.csv")
+    Team.all_teams.each do |team_object|
+      if team_object.team_id == team_id_lowest
```
Comment (zach-bergman) [inline lib/game.rb:155]: Awesome, using the team id to eventually get the team name, which is the return value

**Review by zach-bergman — APPROVED** [review]
Looks great! Merging now

**Pull request merged**